src/test-run/browser-manipulation-queue.js

```diff
@@ -1,75 +1,87 @@
-import { EventEmitter } from 'events';
 import shortId from 'shortid';
-import OS from 'os-family';
-import { resize as resizeWindow, getViewportSize } from 'testcafe-browser-natives';
+import { getViewportSize, generateThumbnail } from 'testcafe-browser-natives';
 import { isServiceCommand } from './commands/utils';
 import COMMAND_TYPE from './commands/type';
-import Warning from '../warnings';
+import WarningsLog from '../warnings/log';
 import WARNING_MESSAGE from '../warnings/message';
 
-export default class BrowserManipulationQueue extends EventEmitter {
-    constructor (windowId, screenshotCapturer) {
-        super();
 
+const SCREENSHOT_MANIPULATION_DESCRIPTION = 'take a screenshot';
+const RESIZE_MANIPULATION_DESCRIPTION     = 'do a window resize';
+
+export default class BrowserManipulationQueue {
+    constructor (browserConnection, screenshotCapturer) {
         this.commands           = [];
-        this.windowId           = windowId;
+        this.windowId           = browserConnection.id;
+        this.browserProvider    = browserConnection.provider;
         this.screenshotCapturer = screenshotCapturer;
+
+        this.warningLog = WarningsLog.get('manipulation:' + this.windowId);
```

Comment:
> What kind of warnings we can get from providers?

>  Also it's logging warning messages using debug package, I found it very useful for myself when debugging providers.

Warnings are things that should go to reports, debug is used for internal debug messages, they shouldn't be mixed